components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -49,6 +65,146 @@ TrackingProtectionService::~TrackingProtectionService() {
   tracking_protection_client_.reset();
 }
 
+#if BUILDFLAG(BRAVE_STP_ENABLED)
+TrackingProtectionService::RenderFrameIdKey::RenderFrameIdKey()
+    : render_process_id(content::ChildProcessHost::kInvalidUniqueID),
+      frame_routing_id(MSG_ROUTING_NONE) {}
+
+TrackingProtectionService::RenderFrameIdKey::RenderFrameIdKey(
+    int render_process_id,
+    int frame_routing_id)
+    : render_process_id(render_process_id),
+      frame_routing_id(frame_routing_id) {}
+
+bool TrackingProtectionService::RenderFrameIdKey::operator<(
+    const RenderFrameIdKey& other) const {
+  return std::tie(render_process_id, frame_routing_id) <
+         std::tie(other.render_process_id, other.frame_routing_id);
+}
+
+bool TrackingProtectionService::RenderFrameIdKey::operator==(
+    const RenderFrameIdKey& other) const {
+  return render_process_id == other.render_process_id &&
+         frame_routing_id == other.frame_routing_id;
+}
+
+void TrackingProtectionService::SetStartingSiteForRenderFrame(
+                                                         GURL starting_site,
+                                                         int render_process_id,
+                                                         int render_frame_id) {
+  base::AutoLock lock(frame_starting_site_map_lock_);
+  const RenderFrameIdKey key(render_process_id, render_frame_id);
+  auto iter = render_frame_key_to_starting_site_url.find(key);
+  if (iter != render_frame_key_to_starting_site_url.end()) {
+    render_frame_key_to_starting_site_url.erase(key);
+  }
+  render_frame_key_to_starting_site_url.insert(
+    std::pair<RenderFrameIdKey, GURL>(key, starting_site));
```

Comment:
instead of erase/insert just use a one-liner `render_frame_key_to_starting_site_url[key] = starting_site`
also no need to `return` from a void method